src/main/java/com/redhat/pantheon/servlet/ModuleRenderServlet.java

```diff
@@ -0,0 +1,121 @@
+
+import com.redhat.pantheon.model.module.Metadata;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.model.module.Content;
+import com.redhat.pantheon.model.module.ModuleRevision;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+
+import org.jetbrains.annotations.NotNull;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.conf.GlobalConfig.CONTENT_TYPE;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+
+/**
+ * Get operation to render a Module data in JSON format.
+ * Only two parameters are expected in the Get request:
+ * 1. locale - Optional; indicates the locale that the module content is in, defaulted to en-US
+ * 2. module_id - indicates the uuid string which uniquely identifies a module
+ *
+ * The url to GET a request from the server is /api/module/render
+ * Example: <server_url>/api/module/render?locale=en-us&module_id=xyz
+ * The said url is accessible outside of the system without any authentication.
+ *
+ * @author Ankit Gadgil
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet to facilitate GET operation which accepts locale and module uuid to output module data",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletPaths(value = "/api/module/render")
+public class ModuleRenderServlet extends AbstractJsonSingleQueryServlet {
+    private final Logger log = LoggerFactory.getLogger(ModuleRenderServlet.class);
+
+    @Override
+    protected String getQuery(SlingHttpServletRequest request) {
+        // Get the query parameter(s)
+        String localeParam = ServletUtils.paramValue(request, "locale", DEFAULT_MODULE_LOCALE.toString());
+        String uuidParam = paramValue(request, "module_id", "");
+
+        StringBuilder query = new StringBuilder("select * from [pant:module]")
+                .append(" as module WHERE ")
+                // look for a specific uuid for module
+                .append("module.[jcr:uuid] = '" + uuidParam + "'");
+        return query.toString();
+    }
+
+    @Override
+    protected Map<String, Object> resourceToMap(@NotNull Resource resource) {
+        Module module = resource.adaptTo(Module.class);
+        // The DEFAULT_MODULE_LOCALE should later be replaced with 'localeParam' variable
+        // this needs to be done while handling localizations
+        Optional<Metadata> draftMetadata = module.getDraftMetadata(DEFAULT_MODULE_LOCALE);
```

Comment:
Sure, there was no specific way for me to choose DEFAULT_MODULE_LOCALE
I was considering not transforming the DEFAULT_MODULE_LOCALE into a locale param till we get to localizations. However, this can make us ready for those changes.

Reply:
The only major reason I did not do this is that the localeParam is local to the getQuery method and we would have to export those to the be used in the Map

Reply:
Ah I see, I didn't notice that `localeParam` was a local variable for the `getQuery` method only. Here's one suggestion - perhaps in the `AbstractJsonSingleQueryServlet`, you store the locale in a private field in the `doGet` method, and expose it via a `protected final String getLocale()` method or something like that. Then the `resourceToMap` method can call it whenever necessary.

Anyway, just a thought, feel free to take it or leave it.

You are right that we don't support localizations yet, but in the cases where it's easy to add future support (like here), we should go ahead and do it.

Reply:
While creating the API fro modules I came across this issue of using locale=en_us as a query param. The code checks for existing locale param from the API url and if not present uses the DEFAULT_MODULE_LOCALE

While converting the the locale string to a Locale object:

localeParam = new Locale(ServletUtils.paramValue(request, "locale", DEFAULT_MODULE_LOCALE.toString()));

While adding as a Locale object the locale variable (string) is converted to a lowercase string by Java:  https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html
This results in a mismatch on the locale for released and draft revisions for a module. Sling is not able to find the releasedRevision for the module and the API throws a 404

Proposed Solution: We need to change the way we store locale as a node.
The typography on the locale node will have to be a lower cased locale, example en_usand not en_US